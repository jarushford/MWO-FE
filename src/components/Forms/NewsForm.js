import React, { Component } from 'react'
import PropTypes from 'prop-types'
import '../../main.scss'

export default class NewsForm extends Component {
  constructor() {
    super()
    this.state = {
      title: '',
      body: '',
      link: ''
    }
  }

  updateValue = (e, input) => {
    this.setState({
      [input]: e.target.value
    })
  }

  handleUpload = e => {
    e.preventDefault()
    window.cloudinary.openUploadWidget(
      { cloud_name: 'mad-wallace', upload_preset: 'xscksodf', tags: [] },
      async (error, result) => {
        if (result.info.secure_url) {
          await this.addPhoto({
            image_url: result.info.secure_url,
            title: this.state.title,
            body: this.state.body,
            link: this.state.link
          })
          this.setState({ 
            title: '',
            body: '',
            link: ''
          })
        }
      }
    )
  }

  addPhoto = async (data) => {
    const url = 'https://mwo-be.herokuapp.com/api/v1/news'
    await fetch(url, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(data)
    })
  }
  
  render() {
    const { showForm } = this.props
    const { title, body, link } = this.state

    return (
      <form
        className={`form ${showForm && 'visible'}`}
        onSubmit={this.handleSubmit}
      >
        <input
          type="text"
          placeholder="title"
          value={title}
          onChange={e => this.updateValue(e, 'title')}
        />
        <input
          type="text"
          placeholder="body"
          value={body}
          onChange={e => this.updateValue(e, 'body')}
        />
        <input
          type="text"
          placeholder="link"
          value={link}
          onChange={e => this.updateValue(e, 'link')}
        />
        <button onClick={this.handleUpload}>Upload & Submit</button>
      </form>
    )
  }
}

NewsForm.propTypes = {
  showForm: PropTypes.bool.isRequired
}