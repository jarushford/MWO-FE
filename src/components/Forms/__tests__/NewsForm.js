import React from 'react'
import { shallow } from 'enzyme'
import NewsForm from '../NewsForm'

describe('NewsForm', () => {
  let wrapper
  
  beforeEach(() => {
    wrapper = shallow(<NewsForm showForm={true} />)
  })

  it('should match the snapshot when showing', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot when not showing', () => {
    const wrapper = shallow(<NewsForm showForm={false} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should have the correct default state', () => {
    const expected = {
      title: '',
      body: '',
      link: ''
    }

    expect(wrapper.state()).toEqual(expected)
  })

  it('updateValue should set the proper value in state', () => {
    const e = { target: { value: 'hi' } }

    wrapper.instance().updateValue(e, 'title')

    expect(wrapper.state().title).toEqual('hi')
  })
})