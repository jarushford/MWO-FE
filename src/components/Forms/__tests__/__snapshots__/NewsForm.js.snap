// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsForm should match the snapshot when not showing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsForm
    showForm={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="title"
          type="text"
          value=""
        />,
        <input
          onChange={[Function]}
          placeholder="body"
          type="text"
          value=""
        />,
        <input
          onChange={[Function]}
          placeholder="link"
          type="text"
          value=""
        />,
        <button
          onClick={[Function]}
        >
          Upload & Submit
        </button>,
      ],
      "className": "form false",
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "title",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "body",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "link",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Upload & Submit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Upload & Submit",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="title"
            type="text"
            value=""
          />,
          <input
            onChange={[Function]}
            placeholder="body"
            type="text"
            value=""
          />,
          <input
            onChange={[Function]}
            placeholder="link"
            type="text"
            value=""
          />,
          <button
            onClick={[Function]}
          >
            Upload & Submit
          </button>,
        ],
        "className": "form false",
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "title",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "body",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "link",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Upload & Submit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Upload & Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`NewsForm should match the snapshot when showing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsForm
    showForm={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="title"
          type="text"
          value=""
        />,
        <input
          onChange={[Function]}
          placeholder="body"
          type="text"
          value=""
        />,
        <input
          onChange={[Function]}
          placeholder="link"
          type="text"
          value=""
        />,
        <button
          onClick={[Function]}
        >
          Upload & Submit
        </button>,
      ],
      "className": "form visible",
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "title",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "body",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "link",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Upload & Submit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Upload & Submit",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="title"
            type="text"
            value=""
          />,
          <input
            onChange={[Function]}
            placeholder="body"
            type="text"
            value=""
          />,
          <input
            onChange={[Function]}
            placeholder="link"
            type="text"
            value=""
          />,
          <button
            onClick={[Function]}
          >
            Upload & Submit
          </button>,
        ],
        "className": "form visible",
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "title",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "body",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "link",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Upload & Submit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Upload & Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
